src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -476,7 +501,15 @@ private void setState(State state) {
   }
 
   private boolean shouldRotate(long now) {
-    boolean periodicRotation = rotateIntervalMs > 0 && now - lastRotate >= rotateIntervalMs;
+    currentTimestamp = timestampExtractor.extract(currentRecord);
+    lastRotate = lastRotate == null && currentRecord != null
+                 ? timestampExtractor.extract(currentRecord)
```

Comment:
Do you mean to call `timestampExtractor.extract(currentRecord)` twice within a few lines?

Reply:
Indeed, but I'm updating the whole block to distinguish between wallclock based and timestamp based partitioning (need to be sure that partitioners that examine the record never operate on `null` records).